packages/commonwealth/server/routes/writeUserSetting.ts

```diff
@@ -22,38 +21,39 @@ const writeUserSetting = async (
   res: Response,
   next: NextFunction,
 ) => {
-  const { key, value } = req.body;
+  const {
+    disable_rich_text,
+    promotional_emails_enabled,
+    email_notification_interval,
+  } = req.body;
 
   if (!req.user) {
     return next(new AppError(Errors.InvalidUser));
   }
-  if (!key || !value) {
-    return next(new AppError(Errors.NoKeyValue));
-  }
 
-  if (key === 'disableRichText' && value === 'true') {
-    req.user.disableRichText = true;
-    await req.user.save();
-  } else if (key === 'disableRichText' && value === 'false') {
-    req.user.disableRichText = false;
-    await req.user.save();
-  } else if (
-    key === 'updateEmailInterval' &&
-    VALID_DIGEST_INTERVALS.indexOf(value) !== -1
-  ) {
-    req.user.emailNotificationInterval = value;
-    await req.user.save();
-  } else if (
-    key === 'promotional_emails_enabled' &&
-    ['true', 'false'].includes(value)
+  if (
+    !(
+      typeof disable_rich_text !== 'boolean' ||
+      typeof promotional_emails_enabled !== 'boolean' ||
+      email_notification_interval
+    )
   ) {
-    req.user.promotional_emails_enabled = value === 'true' ? true : false;
-    await req.user.save();
-  } else {
     return next(new AppError(Errors.InvalidSetting));
   }
 
-  return res.json({ status: 'Success', result: { key, value } });
+  if (typeof disable_rich_text === 'boolean') {
```

Comment:
This stuff should be validated ahead of time and an error thrown earlier if the type is incorrect. If the conditionals are set up correct then Typescript should be able to automatically narrow down the type to Boolean.